src/preact/utils.js

```diff
@@ -49,3 +55,33 @@ export function useResourcesNotify() {
     }
   });
 }
+
+/**
+ * Experimental hook to sync a state based on a property. The state is:
+ * - Initialized from the property;
+ * - Updated independently of the property when property is unchaged;
+ * - Reset to the new property value when the property is updated.
+ * @param {S|function():S} prop
+ * @return {{0: S, 1: function((S|function(S):S)):undefined}}
+ * @template S
+ */
+export function useStateFromProp(prop) {
```

Comment:
I think we need to drop this. I tried it at first, but since then I think this is a wrong pattern and instead we should do a version of controlled-vs-uncontrolled that we tried for selector/accordion.

Reply:
Replaced this with `useState` for now, let's prioritize implementing the controlled/uncontrolled pattern after this initial PR is done.